torch/csrc/distributed/c10d/comm.h

```diff
@@ -51,7 +55,11 @@ struct TORCH_API CommHookInterface {
 
 // PythonCommHook enables registering a python hook to c10d reducer and is a
 // sub class of CommHookInterface.
+#ifdef _WIN32
+class PythonCommHook : public CommHookInterface {
+#else
 class TORCH_API PythonCommHook : public CommHookInterface {
+#endif
```

Comment:
I'm seeing this pattern a lot in the follow-up, why can't we just define TORCH_API to be a no-op on Windows so that we can make one localized change instead of adding this everywhere?